src/BaseFirestoreRepository.ts

```diff
@@ -155,18 +155,28 @@ export default class BaseFirestoreRepository<T extends IEntity>
     await this.firestoreCollection.doc(id).delete();
   }
 
+  limit(limit: number): Promise<T[]> {
+    return this.firestoreCollection
```

Comment:
Since `limit` is just a way to constrain the number of documents that will be retrieved, I wouldn't trigger a query but return a query builder.

Something like this:
`return new QueryBuilder<T>(this).limit(limit);`

And nitpicking a bit, I would not name the parameter of that function limit (maybe limitVal,val, number...) so I don't repeat `limit` twice in the same line 😄

Reply:
I'm assuming the return type of `limit` should become `QueryBuilder<T>`, correct?
See my latest changes.